src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -359,6 +365,27 @@ def get_package_size(self, output):
             return Constants.UNKNOWN_PACKAGE_SIZE
     # endregion
 
+    # region auto OS updates
+    def get_current_auto_os_update_settings(self):
+        """Get current auto OS updates behaviour set on the machine using the command input"""
+        self.composite_logger.log_debug("Fetching current system settings for auto OS updates")
+        try:
+            return self.invoke_package_manager(self.find_current_auto_os_updates_settings_cmd)
+        except Exception as error:
+            self.composite_logger.log_error("Error fetching current system settings for auto OS updates. [Error={0}]".format(str(error)))
+            return None
+
+    def disable_auto_os_update(self):
+        """ Disables auto OS updates on the machine only if they are enabled and logs the default settings the machine comes with """
+        self.log_default_auto_os_updates()
+        self.composite_logger.log_debug("Disabling auto OS updates if they are enabled")
+        try:
+            self.invoke_package_manager(self.disable_auto_os_update_command)
+            self.composite_logger.log("Successfully disabled auto OS updates")
+        except Exception as error:
+            self.composite_logger.log_error("Could not disable auto OS updates.")
```

Comment:
Reminder to bring this discussion up in scrum. Fail fully or not - deliberate choice.